File: Two_Sum/two_sum.cpp
#include <iostream>
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m{};
        vector<int> ans{};
        for(int i = 0; i < nums.size(); i++)
        {
            if(m[target - nums[i]])
            {
                ans = {m[target - nums[i]]-1, i};
            }
            m[nums[i]] = i+1;
        }
        return ans;
    }
};

int main(){
    Solution s{};
    vector<int> nums{2, 7, 11, 15};
    int target = 9;
    vector<int> result = s.twoSum(nums, target);
    cout << "Indices: " << result[0] << ", " << result[1] << endl;
    cout << "Values: " << nums[result[0]] << ", " << nums[result[1]] << endl;

    return 0;
}

File: Two_Sum/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        target_dict = {}

        for i, num in enumerate(nums):
            
            if num in target_dict: return [target_dict[num], i]

            target_dict[target-num] = i



# Example usage
if __name__ == "__main__":
    nums = [2, 7, 11, 15]
    target = 9
    solution = Solution()
    result = solution.twoSum(nums, target)
    print(result)  # Output: [0, 1]